### PR TITLE
boards: add openocd support for nucleo_l476rg and stm32l496g_disco

### PR DESCRIPTION
**Pull request by arthurpi**
The current zephyr sdk now includes openocd configurations for l4
boards, so we can now flash/debug these boards using the zephyr
makefiles

Signed-off-by: Arthur SFEZ <arthur.sfez@gmail.com>

### CODE REVIEW

File: boards/arm/stm32l496g_disco/doc/stm32l496g_disco.rst [hunk 1]
```diff
@@ -172,8 +172,8 @@ Default Zephyr Peripheral Mapping:
 System Clock
 ------------
 
-STM32L496G Discovery System Clock could be driven by internal or external oscillator,
-as well as main PLL clock. By default System clock is driven by PLL clock at 80MHz,
+STM32L496G Discovery System Clock could be driven by an internal or external oscillator,
```
Comment (mbolivar) [inline boards/arm/stm32l496g_disco/doc/stm32l496g_disco.rst]: Fixing the grammar in this paragraph is a good change, but the Zephyr policy is to split commits up into single, logical changes, as described in the contribution guide:

http://docs.zephyrproject.org/contribute/contribute_guidelines.html#commit-guidelines

Since this commit is about openocd support, could you split this change up into a different commit?

File: boards/arm/stm32l496g_disco/pinmux.c [hunk 2]
```diff
@@ -42,4 +42,4 @@ static int pinmux_stm32_init(struct device *port)
 }
 
 SYS_INIT(pinmux_stm32_init, PRE_KERNEL_1,
-	 CONFIG_PINMUX_STM32_DEVICE_INITIALIZATION_PRIORITY);
+		CONFIG_PINMUX_STM32_DEVICE_INITIALIZATION_PRIORITY);
```
Comment (mbolivar) [inline boards/arm/stm32l496g_disco/pinmux.c]: Same here, could you put this in its own commit as a whitespace fix unrelated to openocd support?

**Review by mbolivar — CHANGES_REQUESTED** [review]
Nice cleanup! Thanks for taking the time to look into other places where openocd support was available but not provided.

I have a couple of minor comments.

**Comment by rtur0** [comment]
@mbolivar alright I've split my changes into several commits now !

**Review by dbkinder — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]

File: boards/arm/stm32l496g_disco/pinmux.c (hunk 2 above)
Comment (erwango) [inline boards/arm/stm32l496g_disco/pinmux.c]: I think this change is not needed if you use 8 characters tabs as indicated in coding style guidelines:
http://docs.zephyrproject.org/contribute/contribute_guidelines.html#coding-style
then
https://www.kernel.org/doc/html/latest/process/coding-style.html#indentation,

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
Minor change rested. Otherwise, Thanks for this clean up!

**Comment by mbolivar** [comment]
> Minor change rested. Otherwise, Thanks for this clean up!

+1, LGTM other than this

**Comment by rtur0** [comment]
Oops, I've made a mess with the commits D: Don't mind this.

File: boards/arm/nucleo_l476rg/Makefile.board [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FLASH_SCRIPT = openocd.sh
+DEBUG_SCRIPT = openocd.sh
+
+OPENOCD_LOAD_CMD = "flash write_image erase ${O}/${KERNEL_BIN_NAME} ${CONFIG_FLASH_BASE_ADDRESS}"
+OPENOCD_VERIFY_CMD = "verify_image ${O}/${KERNEL_BIN_NAME} ${CONFIG_FLASH_BASE_ADDRESS}"
+
+export FLASH_SCRIPT OPENOCD_LOAD_CMD OPENOCD_VERIFY_CMD
```
Comment (SebastianBoe) [inline boards/arm/nucleo_l476rg/Makefile.board]: This needs to be ported to CMake.

Doing that involves renaming this file to board.cmake and replacing it's contents with

"include($ENV{ZEPHYR_BASE}/boards/common/openocd.board.cmake)"

The same with the other board in this PR.

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

**Comment by mbolivar** [comment]
> This needs to be ported to CMake.

So close @rtur0! Just one more hurdle.

File: boards/arm/nucleo_l476rg/Makefile.board (hunk 3 above)
Comment (rtur0) [reply boards/arm/nucleo_l476rg/Makefile.board]: Alright I've done this and tested for both boards, this works on my end.

**Review by SebastianBoe — APPROVED** [review]

**Review by galak — APPROVED** [review]
